dynamite_flask_app/create_update_database.py

```diff
@@ -0,0 +1,188 @@
+    for product in products:
+        product_details = Product(
+            product_id=product['id'], name=product['name'], price=product['price'], sale_price=product['sale_price'],
+            description=product['description']
+        )
+
+        add_image_url(product, product_details)
+        add_feature(product, product_details)
+        add_category(product_details)
+        add_size(product, product_details)
+
+        db.session.add(product_details)
+
+    db.session.commit()
+
+
+def add_image_url(product, product_details):
+    """Adds image urls to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for image_url in product['img_urls']:
+        url = ImageUrl(url=image_url)
+        product_details.img_urls.append(url)
+
+
+def add_feature(product, product_details):
+    """adds features to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for detail in product['details']:
+        feature = Feature.query.filter_by(feature=detail).first()
+        if feature is None:
+            feature = Feature(feature=detail)
+        product_details.features.append(feature)
+
+
+def add_size(product, product_details):
+    """adds sizes to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for each_size in product['sizes']:
+        each_size = each_size.upper()
+        available = ProductSize(availability=bool(random.getrandbits(1)))
+
+        size = Size.query.filter_by(size=each_size).first()
+
+        if size is None:
+            size = Size(size=each_size)
+
+        available.size = size
+        available.product = product_details
+
+
+def add_category(product_details):
+    """adds categories to the database.
+
+    Args:
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    category = Category.query.filter_by(type='General').first()
+    if category is None:
+        category = Category()
+    product_details.categories.append(category)
+
+
+def add_user():
+    """adds users to the database."""
+    for user in users:
+        add_new_user(user['name'], user['email'],
+                     user['password'], user['address'], user['contact_no'])
+
+    db.session.commit()
+
+
+def add_order(user, product):
+    """adds orders to the database.
+
+    Args:
+        user(int): The user ID.
+        product(int): The product ID
+
+    """
+    user_id = User.query.get(user)
```

Comment:
this is a user object not an id. but the parameter is an id. the naming should be reversed.

Reply:
done